### DISCUSSION TITLE
Service Discovery support for communication between hypervisor host and guest OS

### DISCUSSION THREAD

**Discussion by huntjw** [eclipse-iceoryx/iceoryx#1818] [Ideas]
I would like to run iceoryx to communicate between a hypervisor guest OS and host OS. The shared memory communication can work because the hypervisor supports shared memory between the guest and host OS. The problem that I have is that Service Discovery uses Unix Domain Sockets, which does not work between the guest and host OS. 

I propose an option for Service Discovery using IP. Then, I can run Roudi in the host OS and applications running in the guest OS will be able to perform Service Discovery over IP using virtual networking.

**Comment by elfenpiff** [comment]
@huntjw I think you can get it running even instantly. For windows we use an IPC mechanism we called NamedPipe which in the end creates another shared memory segment with a queue in there. Could you adjust the alias in
`iceoryx_posh/include/iceoryx_posh/internal/runtime/ipc_interface_base.hpp:53`
to `using IoxIpcChannelType = iox::posix::NamedPipe;` for the non windows version and tell me if it works?

If it does the next step would be that we move this part into our `iceoryx_platform` and you create then, for your target, a new platform which configures exactly this IPC channel. This would even run much faster and efficient then IP and virtual networking.

**Reply by huntjw** [reply]
@elfenpiff Thanks for the brilliant suggestion! I am using version 2.0 for compatibility with cyclonedds. In that version, `IoxIpcChannelType` is already defined in `iceoryx_platform`.

It works fine in Linux with `using IoxIpcChannelType = iox::posix::NamedPipe;` so I think this is the path forward. It would be great to move  this back to `iceoryx_platform` in the latest version to facilitate this use case.

Thanks for your help!
